src/app/icd/ICDManager.cpp

```diff
@@ -123,8 +123,9 @@ void ICDManager::UpdateOperationState(OperationalState state)
 
     if (state == OperationalState::IdleMode)
     {
-        mOperationalState         = OperationalState::IdleMode;
-        uint32_t idleModeInterval = IcdManagementServer::GetInstance().GetIdleModeInterval();
+        mOperationalState = OperationalState::IdleMode;
+        uint32_t idleModeInterval =
+            (IcdManagementServer::GetInstance().GetIdleModeInterval() * 1000); // convert IdleModeInterval from s to ms
         DeviceLayer::SystemLayer().StartTimer(System::Clock::Timeout(idleModeInterval), OnIdleModeDone, this);
```

Comment:
I would suggest just using `System::Clock::Seconds32(idleModeInterval)` in the StartTimer call so the code documents itself, and avoids time arithmetic.

Reply:
done